src/availability/data_source.rs

```diff
@@ -15,51 +15,105 @@ use super::query_data::{
     TransactionIndex,
 };
 use crate::{Block, Deltas, Resolvable};
+use async_trait::async_trait;
 use futures::stream::Stream;
 use hotshot_types::traits::{
     node_implementation::{NodeImplementation, NodeType},
     signature_key::EncodedPublicKey,
 };
+use serde::{Deserialize, Serialize};
+use snafu::Snafu;
 use std::error::Error;
-use std::fmt::Debug;
+use std::fmt::{self, Debug, Display, Formatter};
+use std::ops::RangeBounds;
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum ResourceId<H> {
+    Number(usize),
+    Hash(H),
+}
+
+impl<H: Display> Display for ResourceId<H> {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self {
+            Self::Number(n) => write!(f, "{n}"),
+            Self::Hash(h) => write!(f, "{h}"),
+        }
+    }
+}
```

Comment:
What do you think about adding 

```
impl<H> From<usize> for ResourceId<H> {
    fn from(n: usize) -> Self {
        Self::Number(n)
    }
}
```

to reduce a bit of boilerplate?

Reply:
Good suggestion, will do